python/llm/src/ipex_llm/transformers/convert.py

```diff
@@ -327,9 +327,13 @@ def _replace_with_low_bit_linear(model, qtype, modules_to_not_convert=None,
             optimize_lm_head = False
             if is_lm_head(name, model_config, out_features):
                 model_type = getattr(model_config, "model_type", None)
-                if model_type in ["gptj", "llama", "qwen2"] and \
-                        os.environ.get("IPEX_LLM_LAST_LM_HEAD", None) == "1":
-                    optimize_lm_head = True
+                if model_type in ["gptj", "llama", "qwen2"]:
+                    if os.environ.get("IPEX_LLM_LAST_LM_HEAD", None) is not None:
+                        optimize_lm_head = os.environ.get("IPEX_LLM_LAST_LM_HEAD", None) == "1"
+                    elif os.environ.get("IPEX_LLM_LOW_MEM", None) is not None:
+                        optimize_lm_head = os.environ.get("IPEX_LLM_LOW_MEM", None) == "1"
```

Comment:
We should disable `optimize_lm_head` when using speculative or lookahead.

Reply:
We already have conditions to avoid impacting speculative . But magic value 10 maybe not a good value. in https://github.com/intel-analytics/ipex-llm/blob/51f2effb05108cf459caa76c7e00c2899a5af82c/python/llm/src/ipex_llm/transformers/low_bit_linear.py#L343-L346